src/Text/Parser/Core.idr

```diff
@@ -84,6 +86,7 @@ Functor (Grammar tok c) where
   map f (SeqEmpty act next)
       = SeqEmpty act (\val => map f (next val))
   map {c} f (Bounds act) = rewrite sym $ orFalseNeutral c in SeqEmpty (Bounds act) (Empty . f) -- Bounds (map f act)
+  map {c} f (WeakBounds act) = rewrite sym $ orFalseNeutral c in SeqEmpty (Bounds act) (Empty . f) -- Bounds (map f act)
```

Comment:
Should this preserve `WeakBounds`?

Reply:
Thanks for pointing this out. It was my bad that I just copied and pasted the code from the `Bounds` case. Actually, after reading how `map` is implemented, I realized `map` is exactly what I need to get rid of the position information in some cases. So I just removed the whole `WeakBound` thing and use `map` to get the job done.